auslib/test/admin/views/test_releases.py

```diff
@@ -557,7 +646,14 @@ def testLocalePutForNewRelease(self):
         }
     }
 }
-"""))
+""")
+        self.assertEqual(ret, expected)
+
+        history_rows = dbo.releases.history.t.select().where(dbo.releases.history.name == "e").execute().fetchall()
+        self.assertEqual(len(history_rows), 3)
+        self.assertEqual(history_rows[0]["data"], None)
+        self.assertEqual(history_rows[1]["data"], {"name": "e", "schema_version": 1, "hashFunction": "sha512"})
+        self.assertEqual(history_rows[2]["data"], expected)
```

Comment:
This also seems odd to me - a new product post via the /releases/<release>/builds/[platform]/[locale] endpoint ends up with 3 history rows rather than 2. Would make sense if this was product ab, but it's not here. Am I missing something related to how we test or put/post differences ?

Reply:
I _think_ this is for the same reason as above. Both end up in the changeRelease "helper", and end up at https://github.com/mozilla/balrog/blob/master/auslib/web/admin/views/releases.py#L142 -- which creates the release without the data, and subsequently updates it. So we end up with the empty history row, then data_version 1 & 2 as part of a single http request.

Let me know if there's still any confusion - I know this code/tests are clear as mud.

Reply:
Ah hah - that bit of code explains all my questions, thanks.